streaming/src/main/scala/org/apache/gearpump/streaming/DAG.scala

```diff
@@ -51,6 +52,16 @@ object DAG {
       taskDescription.equals(node)
     }.get._1
   }
+
+  implicit val writer = upickle.Writer[DAG] {
```

Comment:
If some other object contains the DAG, like AppDescription(name, DAG), will uPickle automatically detect the writer you defined here when serializing AppDescription using uPickle?